lib/OpenQA/Scheduler/Model/Jobs.pm

```diff
@@ -34,21 +34,164 @@ use constant MAX_JOB_ALLOCATION => $ENV{OPENQA_SCHEDULER_MAX_JOB_ALLOCATION} //
 has scheduled_jobs  => sub { {} };
 has shuffle_workers => 1;
 
+
+sub _assign_allocated_job_worker_pairs {
+    my ($self, $allocated) = @_;
+
+    my $schema  = OpenQA::Schema->singleton;
+    my $workers = $schema->resultset('Workers');
+
+    # find worker
+    my $worker_id = $allocated->{worker};
+    my $worker;
+    try {
+        $worker = $workers->find({id => $worker_id});
+    }
+    catch {
+        log_debug("Failed to retrieve worker ($worker_id) in the DB, reason: $_");    # uncoverable statement
+    };
+    next unless $worker;
+    if ($worker->unfinished_jobs->count) {
+        log_debug "Worker already got jobs, skipping";
+        next;
```

Comment:
This `next` is not in a `for` or `foreach`, right? does it work here? The same question for the following `next`.